File: app/src/main/java/com/meghanil/ganpatiaarto/MainActivity.java
package com.meghanil.ganpatiaarto;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.widget.NestedScrollView;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.TextView;





public class MainActivity extends AppCompatActivity    {
    NestedScrollView nested;
    public static String TTITTL="tittle";
    public static String DESCRIPTION="description";
    public static  String  IMAGE="image";
    public static  String  POSITION="POSITION";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
        init();
    }

    private void init() {

        Bundle data =getIntent().getExtras();
        String tittle = data.getString(TTITTL,null);
        if(tittle!=null) {
            getSupportActionBar().setTitle(tittle);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }
        String description = data.getString(DESCRIPTION,null);
        if(description!=null){
            ((TextView)findViewById(R.id.description)).setText(description);
        }

        int  imgId = data.getInt(IMAGE,-1);
        if(imgId!=-1) {
        //    Picasso.with(this).load(imgId).into((ImageView) findViewById(R.id.imgID));
        }
        nested = (NestedScrollView) findViewById(R.id.nested);

    }
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();
        if (id == R.id.share) {
            String shareBody = (String) ((TextView)findViewById(R.id.description)).getText();
            Intent sharingIntent = new Intent(android.content.Intent.ACTION_SEND);
            sharingIntent.setType("text/plain");
            sharingIntent.putExtra(Intent.EXTRA_SUBJECT, getSupportActionBar().getTitle());
            sharingIntent.putExtra(android.content.Intent.EXTRA_TEXT, shareBody);
            startActivity(Intent.createChooser(sharingIntent, null));

            return true;
        }
        return super.onOptionsItemSelected(item);
    }



}

File: app/src/main/java/com/meghanil/ganpatiaarto/MainActivity.kt
package com.meghanil.ganpatiaarto

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.arrtilist_item.view.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        init()
    }

    private fun init() {
        val data = intent.extras
        val description1 = data!!.getString(Util.DESCRIPTION, null)
        if (description1 != null) {
            description.text = description1
        }
        val titleTxt = data.getString(Util.TITLE, null)
        if (titleTxt != null) {
            titleView.text = titleTxt
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        if (id == R.id.share) {
            val shareBody = (findViewById<View>(R.id.description) as TextView).text as String
            val sharingIntent = Intent(Intent.ACTION_SEND)
            sharingIntent.type = "text/plain"
            sharingIntent.putExtra(Intent.EXTRA_SUBJECT, supportActionBar!!.title)
            sharingIntent.putExtra(Intent.EXTRA_TEXT, shareBody)
            startActivity(Intent.createChooser(sharingIntent, null))
            return true
        }
        return super.onOptionsItemSelected(item)
    }


}